orttraining/orttraining/python/training/onnxblock/model_accessor.py

```diff
@@ -24,43 +28,90 @@ def model(self):
             )
         return self._model
 
-    @property
-    def eval_model(self):
-        """ModelAccessor property that gets the eval model."""
-
-        if self._eval_model is None:
-            raise RuntimeError("The eval onnx model was not set.")
-        return self._eval_model
-
-    @eval_model.setter
-    def eval_model(self, value):
-        """ModelAccessor property that sets the eval model."""
-        self._eval_model = value
-
 
 # This variable resides in the global namespace.
 # Different methods can access this global model and manipulate it.
-# Its construction and destruction is managed by the onnx_model contextmanager
-global_accessor = None
+# Its construction and destruction is managed by the base and empty_base contextmanagers
+_GLOBAL_ACCESSOR = None
 
 
 @contextmanager
-def onnx_model(model=None):
-    """Context manager that is the entry point to graph manipulations on model.
+def base(model):
+    """Registers the base model to be manipulated by the onnx blocks.
+
+    Example:
+    >>> with onnxblock.base(model) as model_handle:
+    >>>     # manipulate the model using blocks
+    >>>     ...
+    >>>     # get the modified model
+    >>>     model = model_handle.model
+
+    In this example, base will register the given input model to be manipulated by the onnx blocks.
+
+    Args:
+        model (onnx.ModelProto): The base model to be manipulated by the onnx blocks.
```

Comment:
```suggestion
        model: The base model to be manipulated by the onnx blocks.
```

consider adding type to the signature?